libp2p/Host.cpp

```diff
@@ -550,14 +550,22 @@ void Host::run(boost::system::error_code const&)
 	// is always live and to ensure reputation and fallback timers are properly
 	// updated. // disconnectLatePeers();
 
-	if (peerCount() < m_idealPeerCount)
+	auto openSlots = m_idealPeerCount - peerCount();
+	if (openSlots > 0)
 	{
-		for (auto p: m_peers)
-			if (p.second->shouldReconnect())
-			{
-				connect(p.second);
+		list<shared_ptr<Peer>> toConnect;
```

Comment:
I don't know how many results will be inserted ahead of time and expect container to have copy.